DR2-1167 ingest folder opex creator

MancunianSam opened the pull request:
This creates an opex file for a folder. The folder id is passed as input
to the lambda.

techncl commented on src/test/scala/uk/gov/nationalarchives/LambdaTest.scala:
```diff
@@ -0,0 +1,236 @@
+    expectedRequestBody should equal(requestBody)
+  }
+
+  "handleRequest" should "upload the opex file" in {
```
This is only testing the path; do you think:

1. this title should be changed to "upload the opex file using the correct url" or something like that
2. it's worth adding a test to see if the correct opex is sent (either here or a brand new test)?

MancunianSam replied:
I've changed the title of this test and I've added another one in which checks what is sent to S3.

techncl reviewed (commented):
Looks good, just one comment plus these 2:

1. I noticed that there are dynamo request checks but there are no checks for what we are sending to the s3 `.upload` and `.headObject` methods
2. For the XMLCreatorTest, do you think we should add a test for the if conditions, so:
    i. if title is blank
    ii. is not `hierarchyFolder` (i.e. title and name are the same)

MancunianSam commented:
I've added in a test for checking the body uploaded to S3. The head object only gets the file size so that's checked in the same new test. 
I've added in another couple of tests for the XMLCreator as you've suggested.

techncl reviewed (approved):
👍
